### DISCUSSION TITLE
Pro Micro Flashing Issue

### DISCUSSION THREAD

**Discussion by ddulic** [manna-harbour/miryoku#255] [Q&A] [answered]
Heya, I am having an issue flashing a Ferris Sweep MX. I have the [swapped pins](https://github.com/keyboard-magpie/qmk_firmware/tree/mbuk-ferris/keyboards/ferris/sweep_mx_mbuk#swapped-pins) variant.

I have set the handedness on both sides.

I have to first flash the original firmware linked [here](https://github.com/mechboardsguides/mbuk-ferris-sweep-readme) on both sides and then proceed with flashing the miryoku build (with the swapped pins config added) only on the left (main) one. If I don’t do this, the keymaps are all over the place and the keyboard is unusable.

I can also subsequently flash the firmware on the same left side without issues.

My question is, why do I need to flash the original firmware at all? From what I can see the whole controller is getting erased on each flash so it shouldn’t matter. Am I doing something wrong or have potentially missed a step?

I have it working now, this is just for my curiosity.

Thanks!

**Comment by manna-harbour** [comment, marked as answer]
You will need to set handedness if the keyboard definition requires it e.g. https://github.com/keyboard-magpie/qmk_firmware/tree/mbuk-ferris/keyboards/ferris/sweep_choc_mbuk#setting-handedness. But other than that it shouldn't matter which firmware was flashed before. Although, there are occasionally issues with settings that are stored in EEPROM and aren't cleared by flashing, such as the default layer setting, requiring an EEPROM reset to fix.
